README.md

```diff
@@ -114,3 +116,66 @@ func Handle(req handler.Request) (handler.Response, error) {
 	}, err
 }
 ```
+
+## go-middleware
+
+This template uses the http.HandlerFunc as entry point.
+
+### Status of the template
+
+The template makes use of the OpenFaaS incubator project [of-watchdog](https://github.com/openfaas-incubator/of-watchdog).
+
+### Trying the template
+
+```
+$ faas template pull https://github.com/openfaas-incubator/golang-http-template
+$ faas new --lang golang-middleware <fn-name>
+```
+
+### Example usage
+
+Example writing a json response:
+
+```go
+package function
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+)
+
+func Handle(w http.ResponseWriter, r *http.Request) {
```

Comment:
So how are you going to cache a connection pool etc?

Might be better to have a factory here

```
func MakeHandler() func (w http.ResponseWriter, r *http.Request) {


}
```

Then you can do this:

```
func MakeHandler() func (w http.ResponseWriter, r *http.Request) {
   dbConnection := makeExpensiveCallOnceOnly()

   return func (w http.ResponseWriter, r *http.Request) {
        dbConnection.accessResourceVeryFast()
    }
}
```

With your variant this is harder to do.

Reply:
I would handle caching and connection pooling in the init func.

```
var (
	cache []string
	index int
)

// expensive call runs only once
func init() {
	index = 2
	cache = make([]string, index)
	for i := 0; i < index; i++ {
		time.Sleep(1 * time.Second)
		cache[i] = "test"
	}
}

func Handle(w http.ResponseWriter, r *http.Request) {
	// read request payload
	reqBody, err := ioutil.ReadAll(r.Body)
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
}
```

Reply:
@alexellis I've added an example in the latest commit

Reply:
Is `init` guaranteed to have run ahead of time? What if there is an error getting a resource?

The following would be much more robust as a serverless signature:

```
func MakeHandler() (func (w http.ResponseWriter, r *http.Request), error) {
   dbConnection, dbErr := makeExpensiveCallOnceOnly()

   return func (w http.ResponseWriter, r *http.Request) {
        dbConnection.accessResourceVeryFast()
    }, dbErr
}
```

Reply:
Thanks for adding the example.